terraform/modules/jenkins/security.tf

```diff
@@ -1,36 +1,91 @@
-resource "aws_security_group" "ec2" {
-  name        = "${var.app_name}-ec2-security-group"
-  description = "Controls access to the TDR application EC2 instance"
-  vpc_id      = var.ecs_vpc
+data "aws_ip_ranges" "cloudfront_ranges_global" {
+  regions  = ["global"]
+  services = ["cloudfront"]
+}
+
+data "aws_ip_ranges" "cloudfront_ranges_regional" {
+  regions  = ["ap-northeast-1","ap-northeast-2","ap-south-1","ap-southeast-1","ap-southeast-2","ca-central-1","eu-central-1","eu-west-1","eu-west-2","eu-west-3","sa-east-1","us-east-1","us-east-2","us-west-1","us-west-2"]
+  services = ["cloudfront"]
+}
+
+resource "aws_security_group" "ec2_internal" {
+  name = "${var.app_name}-ec2-security-group-internal"
+  description = "Controls access within our network for Jenkins"
+  vpc_id = var.ecs_vpc
 
   ingress {
     protocol    = "tcp"
     from_port   = 80
     to_port     = 80
-    cidr_blocks = ["85.115.53.202/32"]
+    cidr_blocks = ["${var.elastic_ip_address}/32"]
   }
 
   ingress {
     protocol    = "tcp"
     from_port   = 80
     to_port     = 80
-    cidr_blocks = ["0.0.0.0/0"]
+    cidr_blocks = [var.ecs_vpc_cidr_block]
   }
 
   ingress {
     protocol    = "tcp"
-    from_port   = 443
-    to_port     = 443
-    cidr_blocks = ["0.0.0.0/0"]
+    from_port   = 50000
```

Comment:
Worth a comment explaining why Jenkins needs 50000 to be open.

Reply:
I've added one line in to explain